### PR TITLE
docs: `json_path_exists` function

### LINKED ISSUE #1207: Add doc for `json_path_exists` UDF
https://github.com/GreptimeTeam/greptimedb/pull/4807

### PR DESCRIPTION
**Pull request by CookiePieWw**
## What's Changed in this PR

As title. Waiting for https://github.com/GreptimeTeam/greptimedb/pull/4881 to update the case when `json` is null.

Close #1207

## Checklist

- [x] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

**Review by WenyXu — APPROVED** [review]

File: docs/reference/sql/functions/overview.md [hunk 1]
```diff
@@ -302,6 +302,30 @@ SELECT json_is_object(parse_json('1'));
 +---------------------------------------+
 ```
 
+* `json_path_exists(json, path)` to check whether a path exists in a JSON value.
+
+If the path is invalid, the function will return an error.
+
+If the path or the JSON value is NULL, the function will return a NULL value.
```
Comment (nicecui) [inline docs/reference/sql/functions/overview.md:309]:
```suggestion
If the path or the JSON value is `NULL`, the function will return a `NULL` value.
```

**Review by nicecui — COMMENTED** [review]
What scenarios does the `json_path_exists` solve?

File: docs/reference/sql/functions/overview.md [hunk 2]
```diff
@@ -302,6 +302,30 @@ SELECT json_is_object(parse_json('1'));
...
+
+```sql
+SELECT json_path_exists(parse_json('{"a": 1, "b": 2}'), 'a');
+
++------------------------------------------------------------------+
+| json_path_exists(parse_json(Utf8("{"a": 1, "b": 2}")),Utf8("a")) |
++------------------------------------------------------------------+
+|                                                                1 |
++------------------------------------------------------------------+
```
Comment (WenyXu) [inline docs/reference/sql/functions/overview.md:318]:
```suggestion
+------------------------------------------------------------------+
|                                                            false |
+------------------------------------------------------------------+
```

File: i18n/zh/docusaurus-plugin-content-docs/current/reference/sql/functions/overview.md [hunk 3]
```diff
@@ -295,6 +295,30 @@ SELECT json_is_object(parse_json('1'));
...
+如果 `path` 或 `json` 是 `NULL`，函数将返回 `NULL`。
+
+```sql
+SELECT json_path_exists(parse_json('{"a": 1, "b": 2}'), 'a');
+
++------------------------------------------------------------------+
+| json_path_exists(parse_json(Utf8("{"a": 1, "b": 2}")),Utf8("a")) |
++------------------------------------------------------------------+
+|                                                                1 |
++------------------------------------------------------------------+
```
Comment (WenyXu) [inline i18n/zh/docusaurus-plugin-content-docs/current/reference/sql/functions/overview.md:311]:
```suggestion
+------------------------------------------------------------------+
|                                                            false |
+------------------------------------------------------------------+
```

File: docs/reference/sql/functions/overview.md (hunk 2 above)
Comment (CookiePieWw) [reply docs/reference/sql/functions/overview.md:318]: 0/1 is consistent to previous functions in the docs, do we need to change them all?

https://github.com/GreptimeTeam/docs/blob/24b71920caae49d4d3b45823bea838ee4367dbd6/docs/reference/sql/functions/overview.md?plain=1#L189-L198

Comment (CookiePieWw) [reply docs/reference/sql/functions/overview.md:318]: 0/1 is consistent with previous docs, do we need to change them all?
https://github.com/GreptimeTeam/docs/blob/24b71920caae49d4d3b45823bea838ee4367dbd6/docs/reference/sql/functions/overview.md?plain=1#L189-L198

Comment (WenyXu) [reply docs/reference/sql/functions/overview.md:318]: > docs/docs/reference/sql/functions/overview.md

Oh, I see.

File: i18n/zh/docusaurus-plugin-content-docs/current/reference/sql/functions/overview.md [hunk 4]
```diff
@@ -295,6 +295,30 @@ SELECT json_is_object(parse_json('1'));
...
+
```
Comment (WenyXu) [inline i18n/zh/docusaurus-plugin-content-docs/current/reference/sql/functions/overview.md:312]:
```suggestion

SELECT json_path_exists(parse_json('{"a": 1, "b": 2}'), 'c.d');

 +------------------------------------------------------------------+
 | json_path_exists(parse_json(Utf8("{"a": 1, "b": 2}")),Utf8("c.d")) |
 +------------------------------------------------------------------+
 |                                                                0 |
 +------------------------------------------------------------------+

```

**Review by WenyXu — APPROVED** [review]
Rest LGTM

**Comment by CookiePieWw** [comment]
> What scenarios does the `json_path_exists` solve?

For me it can test the structure of a json value without caring about its value. . It's a T-SQL function actually.

**Review by nicecui — APPROVED** [review]

**Pull request merged**